mslib/msui/editor.py

```diff
@@ -114,41 +118,78 @@
+        parent_list.append(index)
+    parent_list.reverse()
+    if parents:
+        return index, parent_list
+    return index
+
+
 class JsonDelegate(delegate.JsonDelegate):
 
     def paint(self, painter, option, index):
         """Use method from the data type or fall back to the default."""
         if index.column() == 0:
-            return super(JsonDelegate, self).paint(painter, option, index)
-
-        # Bold the value if not default
-        if index.data() is not None:
-            model = index.model()
-            # Get root parent of the index
-            parents = [model.index(index.row(), 0, index.parent())]
-            while parents[0].parent().isValid():
-                parents.insert(0, parents[0].parent())
-            root_option = parents[0].data()
-            if root_option in fixed_dict_options:
-                default_data = default_options[root_option]
+            source_model = index.model()
+            if isinstance(source_model, QtCore.QAbstractProxyModel):
+                source_model = source_model.sourceModel()
+            data = source_model.serialize()
+
+            # bold the key which has non-default value
+            root_index, parents = get_root_index(index, parents=True)
+            parents.append(index)
+            key = root_index.data()
+            if key in list_option_structure or key in dict_option_structure or key in key_value_options:
+                if root_index == index and data[key] != default_options[key]:
+                    option.font.setWeight(QtGui.QFont.Bold)
+            elif key in fixed_dict_options:
+                model_data = data[key]
+                default_data = default_options[key]
                 for parent in parents[1:]:
                     parent_data = parent.data()
                     if isinstance(default_data, list):
                         parent_data = int(parent.data())
+                    model_data = model_data[parent_data]
                     default_data = default_data[parent_data]
-                if default_data != index.data():
-                    option.font.setWeight(QtGui.QFont.Bold)
-            elif root_option in key_value_options:
-                if default_options[root_option] != index.data():
-                    option.font.setWeight(QtGui.QFont.Bold)
-            elif root_option in dict_option_structure or root_option in list_option_structure:
-                # get source model if sortfilterproxymodel is used, for serializing data
-                if isinstance(model, QtCore.QAbstractProxyModel):
-                    model = model.sourceModel()
-                source_data = model.serialize()
-                if source_data[root_option] != default_options[root_option]:
+                if model_data != default_data:
                     option.font.setWeight(QtGui.QFont.Bold)
 
+            return super(JsonDelegate, self).paint(painter, option, index)
+
+#         # Bold the value if not default
```

Comment:
not sure if this only temporary removed, but such large segments if that not needed can also be removed

Reply:
Forgot to remove those unwanted lines. I'll remove them.